qubekit/fragmentation/WBOFragmenter.py

```diff
@@ -0,0 +1,228 @@
+import itertools
+from typing import List, TYPE_CHECKING
+from typing_extensions import Literal
+
+from pydantic import Field
+from qcelemental.util import which_import
+from openff.toolkit.topology import Molecule as OFFMolecule
+from rdkit import Chem
+
+from qubekit.molecules import Ligand
+from qubekit.utils.datastructures import StageBase
+
+if TYPE_CHECKING:
+    from openff.toolkit.topology import Molecule
+    from openff.fragmenter.fragment import (
+        FragmentationResult,
+    )
+
+
+class WBOFragmenter(StageBase):
+
+    type: Literal["WBOFragmenter"] = "WBOFragmenter"
+
+    rotatable_smirks: List[str] = Field(
+        ["[!#1]~[!$(*#*)&!D1:1]-,=;!@[!$(*#*)&!D1:2]~[!#1]"],
+        description="The rotatable smirks to use for fragmentation",
+    )
+
+    @classmethod
+    def is_available(cls) -> bool:
+        fragmenter = which_import(
+            "openff.fragmenter",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge openff-fragmenter`.",
+        )
+
+        rdkit = which_import(
+            "rdkit",
+            return_bool=True,
+            raise_error=True,
+            raise_msg="Please install via `conda install -c conda-forge rdkit`.",
+        )
+
+        return fragmenter and rdkit
+
+    def _prepare_fragment_input(self, molecule: Ligand):
+        """
+        TODO
+        """
+        # convert to OFF molecule
+        mapped_smiles = molecule.to_smiles(mapped=True)
+        off_molecule: OFFMolecule = OFFMolecule.from_mapped_smiles(mapped_smiles)
+
+        # remember the atom names
+        for off_atom, lig_atom in zip(off_molecule.atoms, molecule.atoms):
+            assert off_atom.atomic_number == lig_atom.atomic_number
+            off_atom.name = lig_atom.atom_name
+
+        # convert to the canonical order which will be used by the fragmenter
+        return off_molecule.canonical_order_atoms()
+
+    def _create_ligand(self, semi_mapped_smile, name, bond_indices=None):
+        """
+        TODO
+        """
+        # sanitize is off to keep the "atom_map" property
+        rdmol = Chem.MolFromSmiles(semi_mapped_smile, sanitize=False)
+
+        # add at least one conformer
+        Chem.AllChem.EmbedMolecule(rdmol)
+
+        # Chem.SanitizeMol calls updatePropertyCache so we don't need to call it ourselves
+        # https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1MolOps.html#a8d831787aaf2d65d9920c37b25b476f5
+        Chem.SanitizeMol(
+            rdmol,
+            Chem.SANITIZE_ALL ^ Chem.SANITIZE_ADJUSTHS ^ Chem.SANITIZE_SETAROMATICITY,
+        )
+        Chem.SetAromaticity(rdmol, Chem.AromaticityModel.AROMATICITY_MDL)
+
+        # Chem.MolFromSmiles adds bond directions (i.e. ENDDOWNRIGHT/ENDUPRIGHT), but
+        # doesn't set bond.GetStereo(). We need to call AssignStereochemistry for that.
+        Chem.AssignStereochemistry(rdmol)
```

Comment:
Is it possible to integrate this with out current `Ligand.from_smiles` logic to duplicate so of the code?

Reply:
This does look similar to `smiles_to_rdkit_mol` in `utils.py`. I'll switch the default to `sanitize=False` since it's done separately and see how that fares.